### PR TITLE
Add authorization management

### PR DESCRIPTION
**Pull request by zannkukai**
Create some guard to protect access to some 'new' form URL :
 * Generic guard based on user role
 * LibraryGuard : Check if the current logged user and it current library can manage the resource.
 * Closes https://github.com/rero/rero-ils/issues/787

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/admin/src/app/guard/library.guard.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  /** Check if the current logged user is linked to the same library than the desired resource.
+   *  If access is denied --> 403 : forbidden
+   *  If error occcured --> 500 : Internal server error
+   */
+  canActivate(next: ActivatedRouteSnapshot, state: RouterStateSnapshot)
+    : Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    this.getOwningLibrary(next).subscribe(
+      (owningLibrary) => {
+        console.log(owningLibrary);
```
Comment (AoNoOokami) [inline projects/admin/src/app/guard/library.guard.ts]: Is this console.log necessary?

File: projects/admin/src/app/guard/role.guard.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ *  { path: 'new', component: MyComponent, canActivate: [RoleGuard], data: {roles: ['xxx', 'yyy']}  }
+ */
+export class RoleGuard implements CanActivate {
+
+  constructor(
+    private _userService: UserService,
+    private _router: Router
+  ) { }
+
+  canActivate(next: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> | boolean {
```
Comment (AoNoOokami) [inline projects/admin/src/app/guard/role.guard.ts]: Missing doc

File: projects/admin/src/app/routes/locations-route.ts [hunk 3]
```diff
@@ -18,6 +18,7 @@ import { BaseRoute } from './Base-route';
 import { CanUpdateGuard } from '../guard/can-update.guard';
 import { RouteInterface, DetailComponent, EditorComponent } from '@rero/ng-core';
 import { LocationDetailViewComponent } from '../record/detail-view/location-detail-view/location-detail-view.component';
+import { LibraryGuard } from '../guard/library.guard';
```
Comment (AoNoOokami) [inline projects/admin/src/app/routes/locations-route.ts]: Not sure that the imports are sorted correctly

File: projects/admin/src/app/guard/library.guard.ts (hunk 1 above)
Comment (zannkukai) [reply projects/admin/src/app/guard/library.guard.ts]: sure... not. Removed !

File: projects/admin/src/app/guard/acq-order-line.guard.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  /** Return the library linked to an acquisitin order number.
+   *  @param route: the current URL route
+   *  @return: the library pid linked to the resource from the 'order' query parameters
+   */
+  getOwningLibrary(route: ActivatedRouteSnapshot): Observable<string> {
+    return this._recordService.getRecord('acq_orders', route.queryParams.order).pipe(
+      map( data => data.metadata || {} ),
+      map( metadata => metadata.library || {} ),
+      map( library => extractIdOnRef(library.$ref) )
```
Comment (sebastiendeleze) [inline projects/admin/src/app/guard/acq-order-line.guard.ts]: I would use one map function that does all the job, instead of 3

**Pull request merged**